src/Hal/Bundle/PhpMetricsCollector/DependencyInjection/Configuration.php

```diff
@@ -4,20 +4,21 @@
 use Symfony\Component\Config\Definition\Builder\TreeBuilder;
 use Symfony\Component\Config\Definition\ConfigurationInterface;
 
-/**
- * This is the class that validates and merges configuration from your app/config files
- *
- * To learn more see {@link http://symfony.com/doc/current/cookbook/bundles/extension.html#cookbook-bundles-extension-config-class}
- */
 class Configuration implements ConfigurationInterface
 {
     /**
      * {@inheritDoc}
      */
     public function getConfigTreeBuilder()
     {
-        $treeBuilder = new TreeBuilder();
-        $rootNode = $treeBuilder->root('phpmetrics');
+        $treeBuilder = new TreeBuilder('phpmetrics');
+        // Keep compatibility with symfony/config < 4.2
+        if (!method_exists($treeBuilder, 'getRootNode')) {
+            $root = $treeBuilder->root('phpmetrics');
+        } else {
+            $root = $treeBuilder->getRootNode();
+        }
+
```

Comment:
No need to store the result in a variable we don't use.

```suggestion
        if (!method_exists($treeBuilder, 'getRootNode')) {
            $treeBuilder->root('phpmetrics');
        } else {
            $treeBuilder->getRootNode();
        }

```